src/coreclr/jit/lsrabuild.cpp

```diff
@@ -3543,6 +3543,7 @@ int LinearScan::BuildReturn(GenTree* tree)
                 noway_assert(op1->IsMultiRegCall() || op1->IsMultiRegLclVar());
 
                 int                   srcCount;
+                ReturnTypeDesc        nonCallRetTypeDesc;
```

Comment:
> Move another local up a scope.

what was the error here?

Reply:
The address of the old `retTypeDesc` was assigned to `pRetTypeDesc`, which was then used after the scope (ends on line 3561) on line 3562